--- udev-025/extras/multipath-tools/Makefile.dm	2004-04-24 21:20:51.037958682 +0200
+++ udev-025/extras/multipath-tools/Makefile	2004-04-24 21:20:56.620034738 +0200
@@ -2,7 +2,7 @@
 #
 # Copyright (C) 2003 Christophe Varoqui, <christophe.varoqui@free.fr>
 
-SUBDIRS = libdevmapper devmap_name multipath multipathd
+SUBDIRS = devmap_name multipath multipathd
 
 recurse:
 	@for dir in $(SUBDIRS); do\
