--- udev-129/udev/lib/libudev-private.h.orig	2008-09-30 23:54:56.000000000 +0200
+++ udev-129/udev/lib/libudev-private.h	2008-10-06 21:58:46.046638174 +0200
@@ -35,9 +35,9 @@
 #  define info(udev, arg...) udev_log(udev, LOG_INFO, __FILE__, __LINE__, __FUNCTION__, ## arg)
 #  define err(udev, arg...) udev_log(udev, LOG_ERR, __FILE__, __LINE__, __FUNCTION__, ## arg)
 #else
-#  define dbg log_null
-#  define info log_null
-#  define err log_null
+#  define dbg(u, f...) log_null(u, f)
+#  define info(u, f...) log_null(u, f)
+#  define err(u, f...) log_null(u, f)
 #endif
 
 /* libudev */
