--- udev-053/namedev.c.orig	2005-02-11 20:55:12.000000000 +0100
+++ udev-053/namedev.c	2005-02-12 10:51:10.614856944 +0100
@@ -779,9 +779,19 @@
 			/* collect symlinks for this or the final matching rule */
 			if (dev->symlink[0] != '\0') {
 				char temp[NAME_SIZE];
+				char *p;
+				int len;
 
 				info("configured rule in '%s[%i]' applied, added symlink '%s'",
 				     dev->config_file, dev->config_line, dev->symlink);
+				foreach_strpart(dev->symlink, " ", p, len) {
+					strncpy(temp, p, len);
+					temp[len] = '\0';
+					apply_format(udev, temp, sizeof(temp), class_dev, sysfs_device);
+					if (udev->symlink[0] != '\0')
+						strfieldcat(udev->symlink, " ");
+					strfieldcat(udev->symlink, temp);
+				}
 				strfieldcpy(temp, dev->symlink);
 				apply_format(udev, temp, sizeof(temp), class_dev, sysfs_device);
 				if (udev->symlink[0] != '\0')
