diff -urN aa/udev-032/extras/start_udev udev-032/extras/start_udev
--- aa/udev-032/extras/start_udev	2004-09-14 07:55:34.000000000 +0200
+++ udev-032/extras/start_udev	2004-09-18 12:19:49.004126920 +0200
@@ -27,6 +27,7 @@
 sysfs_dir=/sys
 bin=/sbin/udev
 udevd=/sbin/udevd
+mknod=/bin/mknod
 
 run_udev () {
 	export ACTION=add
@@ -75,8 +76,41 @@
 	ln -snf /proc/self/fd/2 $udev_root/stderr
 	ln -snf /proc/kcore $udev_root/core
 
-	mkdir $udev_root/pts
-	mkdir $udev_root/shm
+	[ -d $udev_root/pts ] || mkdir $udev_root/pts
+	[ -d $udev_root/shm ] || mkdir $udev_root/shm
+
+	if [ -x $mknod ]; then
+		# fd devices
+		for i in 0 1 2 3 4 5 6 7; do
+			$mknod -m 660 $udev_root/fd$i b 2 $i
+			chown root:floppy $udev_root/fd$i
+		done
+		
+		# loop devices
+		for i in 0 1 2 3 4 5 6 7; do
+			$mknod -m 660 $udev_root/loop$i b 7 $i
+			chown root:disk $udev_root/loop$i
+		done
+		
+		# lp and par devices
+		for i in 0 1 2 3; do
+			$mknod -m 660 $udev_root/lp$i c 6 $i
+			$mknod -m 660 $udev_root/par$i c 6 $i
+			chown root:lp $udev_root/lp$i
+			chown root:lp $udev_root/par$i
+		done
+		
+		# net/tun device
+		mkdir $udev_root/net
+		$mknod -m 600 $udev_root/net/tun c 10 200
+
+		# ppp device
+		$mknod -m 644 $udev_root/ppp c 108 0
+
+		# cpu/microcode device
+		mkdir $udev_root/cpu
+		$mknod -m 666 $udev_root/cpu/microcode c 10 184
+	fi
 }
 
 # don't use udev if sysfs is not mounted.
@@ -84,8 +118,8 @@
 	exit 1
 fi
 
-echo "mounting... ramfs at $udev_root"
-mount -n -t ramfs none $udev_root
+echo "mounting... tmpfs at $udev_root"
+mount -n -o mode=0755 -t tmpfs none $udev_root
 
 # propogate /udev from /sys
 echo "Creating initial udev device nodes:"
