diff -urN aa/udev-032/etc/udev/udev.rules udev-032/etc/udev/udev.rules
--- aa/udev-032/etc/udev/udev.rules	2004-09-14 07:55:34.000000000 +0200
+++ udev-032/etc/udev/udev.rules	2004-09-20 01:25:01.986445600 +0200
@@ -9,7 +9,7 @@
 
 # create a symlink named after the device map name
 # note devmap_name comes with extras/multipath
-KERNEL="dm-[0-9]*", PROGRAM="/sbin/devmap_name %M %m", NAME="%k", SYMLINK="%c"
+#KERNEL="dm-[0-9]*", PROGRAM="/sbin/devmap_name %M %m", NAME="%k", SYMLINK="%c"
 
 # DRI devices always go into a subdirectory (as per the LSB spec)
 KERNEL="card*", NAME="dri/card%n"
@@ -18,7 +18,7 @@
 KERNEL="controlC[0-9]*", NAME="snd/%k"
 KERNEL="hw[CD0-9]*",     NAME="snd/%k"
 KERNEL="pcm[CD0-9cp]*",  NAME="snd/%k"
-KERNEL="midiC[D0-9]*",   NAME="snd/%k"
+KERNEL="midi[CD0-9]*",   NAME="snd/%k"
 KERNEL="timer",          NAME="snd/%k"
 KERNEL="seq",            NAME="snd/%k"
 
@@ -45,3 +45,28 @@
 
 # raw devices
 KERNEL="raw[0-9]*",	NAME="raw/%k" 
+
+KERNEL="microcode",     NAME="%k", SYMLINK="cpu/0/%k"
+KERNEL="ram1", 		NAME="%k", SYMLINK="ram"
+KERNEL="video0",	NAME="%k", SYMLINK="video"
+KERNEL="radio0",	NAME="%k", SYMLINK="radio"
+KERNEL="audio0",	NAME="%k", SYMLINK="audio"
+KERNEL="dsp0",		NAME="%k", SYMLINK="dsp"
+KERNEL="fb0",		NAME="%k", SYMLINK="fb"
+KERNEL="qft0",		NAME="%k", SYMLINK="ftape"
+KERNEL="isdnctrl0",	NAME="%k", SYMLINK="isdnctrl"
+KERNEL="mixer0",	NAME="%k", SYMLINK="mixer"
+KERNEL="ram0",		NAME="%k", SYMLINK="ramdisk"
+KERNEL="sbpcd0",	NAME="%k", SYMLINK="sbpcd"
+KERNEL="radio0",	NAME="%k", SYMLINK="radio"
+KERNEL="tty0",		NAME="%k", SYMLINK="systty"
+KERNEL="vbi0",		NAME="%k", SYMLINK="vbi"
+
+# old compat symlinks with enumeration
+KERNEL="sr*", NAME="%k", SYMLINK="cdrom%e"
+KERNEL="scd*", NAME="%k", SYMLINK="cdrom%e"
+KERNEL="pcd*", NAME="%k", SYMLINK="cdrom%e"
+KERNEL="hd[a-z]", PROGRAM="/bin/cat /proc/ide/%k/media", RESULT="cdrom", NAME="%k", SYMLINK="cdrom%e"
+KERNEL="fd[0-9]", NAME="%k", SYMLINK="floppy%e"
+KERNEL="hd[a-z]", PROGRAM="/bin/cat /proc/ide/%k/media", RESULT="floppy", NAME="%k", SYMLINK="floppy%e
+
