diff -ur udev-111.org/extras/volume_id/lib/Makefile udev-111/extras/volume_id/lib/Makefile
--- udev-111.org/extras/volume_id/lib/Makefile	2007-05-10 12:42:56.000000000 +0200
+++ udev-111/extras/volume_id/lib/Makefile	2007-05-26 12:37:34.079711135 +0200
@@ -4,9 +4,9 @@
 #
 # Released under the GNU General Public License, version 2.
 #
-includedir = 	${prefix}/usr/include
-libdir =	${prefix}/lib
-usrlibdir =	${prefix}/usr/lib
+includedir ?= 	${prefix}/usr/include
+libdir ?=	${prefix}/lib
+usrlibdir ?=	${prefix}/usr/lib
 
 INSTALL = install -c
 INSTALL_DATA  = ${INSTALL} -m 644
diff -ur udev-111.org/extras/volume_id/Makefile udev-111/extras/volume_id/Makefile
--- udev-111.org/extras/volume_id/Makefile	2007-05-10 12:42:56.000000000 +0200
+++ udev-111/extras/volume_id/Makefile	2007-05-26 12:38:38.650029152 +0200
@@ -17,6 +17,9 @@
 libudevdir =	${prefix}/lib/udev
 mandir =	${prefix}/usr/share/man
 configdir =	${etcdir}/udev/
+libdir ?=       ${prefix}/lib
+usrlibdir ?=    ${prefix}/usr/lib
+
 
 INSTALL = install -c
 INSTALL_PROGRAM = ${INSTALL}
@@ -34,7 +37,10 @@
 	$(Q) $(CC) -c $(CFLAGS) $< -o $@
 
 lib:
-	$(Q) $(MAKE) -C $@
+	$(Q) $(MAKE) -C $@ \
+		libdir=$(libdir) \
+		usrlibdir=$(usrlibdir)
+
 .PHONY: lib
 
 lib/libvolume_id.a: lib
diff -ur udev-111.org/Makefile udev-111/Makefile
--- udev-111.org/Makefile	2007-05-10 12:42:56.000000000 +0200
+++ udev-111/Makefile	2007-05-26 12:37:34.079711135 +0200
@@ -98,6 +98,8 @@
 libudevdir =	${prefix}/lib/udev
 mandir =	${prefix}/usr/share/man
 configdir =	${etcdir}/udev
+libdir ?= 	${prefix}/lib
+usrlibdir ?= 	${prefix}/usr/lib
 udevdir =	/dev
 DESTDIR =
 
@@ -168,6 +170,8 @@
 			RANLIB="$(RANLIB)" \
 			LIB_OBJS="$(LIB_OBJS)" \
 			LIBUDEV="$(PWD)/$(LIBUDEV)" \
+			libdir=$(libdir) \
+			usrlibdir=$(usrlibdir) \
 			-C $$target $@ || exit 1; \
 	done;
 .PHONY: all
