--- udev-032/namedev.c.symlink	2004-10-06 10:38:58.760423024 +0200
+++ udev-032/namedev.c	2004-10-06 10:38:58.767421960 +0200
@@ -898,14 +898,18 @@
 
 			if (dev->symlink[0] != '\0') {
 				char temp[NAME_SIZE];
-
+				char *p;
+				int len;
 				info("configured rule in '%s' at line %i applied, added symlink '%s'",
 				     dev->config_file, dev->config_line, dev->symlink);
-				strfieldcpy(temp, dev->symlink);
-				apply_format(udev, temp, sizeof(temp), class_dev, sysfs_device);
-				if (udev->symlink[0] != '\0')
-					strfieldcat(udev->symlink, " ");
-				strfieldcat(udev->symlink, temp);
+				foreach_strpart(dev->symlink, " ", p, len) {
+					strncpy(temp, p, len);
+					temp[len] = '\0';
+					apply_format(udev, temp, sizeof(temp), class_dev, sysfs_device);
+					if (udev->symlink[0] != '\0')
+						strfieldcat(udev->symlink, " ");
+					strfieldcat(udev->symlink, temp);
+				}
 			}
 
 			if (dev->name[0] != '\0') {
