--- Makefile.am	2009-08-15 22:04:56.536696651 +0200
+++ /tmp/Makefile.am	2009-08-15 22:04:31.812740885 +0200
@@ -534,12 +534,13 @@
 
 extras/gudev/GUdev-1.0.gir: extras/gudev/libgudev-1.0.la $(G_IR_SCANNER) Makefile.am
 	$(AM_V_GEN)PKG_CONFIG_PATH=$(top_builddir)/data:$$PKG_CONFIG_PATH \
+	LD_LIBRARY_PATH=extras/gudev/.libs \
 	$(G_IR_SCANNER) -v \
 		--namespace GUdev \
 		--nsversion=1.0 \
 		--include=GObject-2.0 \
 		--library=gudev-1.0 \
-		--library-path=extras/gudev \
+		--library-path=extras/gudev/.libs \
 		--output $@ \
 		--pkg=glib-2.0 \
 		--pkg=gobject-2.0 \
@@ -556,4 +557,5 @@
 		$(top_srcdir)/extras/gudev/gudevdevice.c
 
 extras/gudev/GUdev-1.0.typelib: extras/gudev/GUdev-1.0.gir $(G_IR_COMPILER)
-	$(AM_V_GEN)g-ir-compiler $< -o $@
+	$(AM_V_GEN)LD_LIBRARY_PATH=extras/gudev/.libs \
+	g-ir-compiler $< -o $@
