--- udev-091/Makefile.lib64	2006-04-25 22:39:59.000000000 +0200
+++ udev-091/Makefile	2006-05-03 18:16:31.000000000 +0200
@@ -103,6 +103,8 @@
 libudevdir =	${prefix}/lib/udev
 mandir =	${prefix}/usr/share/man
 configdir =	${etcdir}/udev
+libdir ?= 	${prefix}/lib
+usrlibdir ?= 	${prefix}/usr/lib
 udevdir =	/dev
 DESTDIR =
 
@@ -179,6 +181,8 @@
 			RANLIB="$(RANLIB)" \
 			LIB_OBJS="$(LIB_OBJS)" \
 			LIBUDEV="$(PWD)/$(LIBUDEV)" \
+			libdir=$(libdir) \
+			usrlibdir=$(usrlibdir) \
 			-C $$target $@ || exit 1; \
 	done;
 .PHONY: all
--- udev-091/extras/volume_id/lib/Makefile.lib64	2006-05-03 18:15:28.000000000 +0200
+++ udev-091/extras/volume_id/lib/Makefile	2006-05-03 18:15:53.000000000 +0200
@@ -4,9 +4,9 @@
 #
 # Released under the GNU General Public License, version 2.
 #
-includedir = 	${prefix}/usr/include
-libdir =	${prefix}/lib
-usrlibdir =	${prefix}/usr/lib
+includedir ?= 	${prefix}/usr/include
+libdir ?=	${prefix}/lib
+usrlibdir ?=	${prefix}/usr/lib
 
 INSTALL = /usr/bin/install -c
 INSTALL_DATA  = ${INSTALL} -m 644
--- udev-091/extras/volume_id/Makefile.lib64	2006-04-25 22:39:59.000000000 +0200
+++ udev-091/extras/volume_id/Makefile	2006-05-03 18:16:14.000000000 +0200
@@ -17,6 +17,9 @@
 libudevdir =	${prefix}/lib/udev
 mandir =	${prefix}/usr/share/man
 configdir =	${etcdir}/udev/
+libdir ?=       ${prefix}/lib
+usrlibdir ?=    ${prefix}/usr/lib
+
 
 INSTALL = /usr/bin/install -c
 INSTALL_PROGRAM = ${INSTALL}
@@ -32,7 +35,10 @@
 	$(Q) $(CC) -c $(CFLAGS) $< -o $@
 
 lib/libvolume_id.a:
-	$(Q) $(MAKE) -C lib
+	$(Q) $(MAKE) \
+		libdir=$(libdir) \
+		usrlibdir=$(usrlibdir) \
+		-C lib
 
 $(PROG): %: $(HEADERS) %.o lib/libvolume_id.a
 	$(E) "  LD      " $@
