diff -urN aa/udev-032/etc/udev/udev.permissions udev-032/etc/udev/udev.permissions
--- aa/udev-032/etc/udev/udev.permissions	2004-09-14 07:55:34.000000000 +0200
+++ udev-032/etc/udev/udev.permissions	2004-09-20 01:24:22.852394880 +0200
@@ -1,10 +1,120 @@
-#name:user:group:mode
-ttyUSB0:root:uucp:0660
-ttyUSB1:root:uucp:0666
-pl2303:500:500:0666
-visor:root:uucp:0666
-dsp1:::0666
-camera*:500:500:0666
-sgi_fetchop:root:root:666
-iseries/vcd*:root:disk:660
-iseries/vd*:root:disk:660
+# udev.permissions: permission/ownership map for udev
+# 
+# Syntax of this file:
+#   name:user:group:mode
+
+# console devices
+console:root:console:0660
+tty:root:root:0666
+
+# audio devices
+admmidi*:root:audio:0660
+adsp*:root:audio:0660
+aload*:root:audio:0660
+amidi*:root:audio:0660
+amixer*:root:audio:0660
+audio*:root:audio:0660
+dmfm*:root:audio:0660
+dmmidi*:root:audio:0660
+dsp*:root:audio:0660
+midi*:root:audio:0660
+mixer*:root:audio:0660
+sequencer*:root:audio:0660
+sndstat:root:root:0666
+snd/*:root:audio:0660
+
+# optical devices
+cdroms/*:root:disk:0660
+pcd*:root:disk:0660
+scd*:root:disk:0660
+
+# disk devices
+ataraid*:root:disk:0660
+discs/*/*:root:disk:0660
+hd*:root:disk:0660
+loop*:root:disk:0660
+md*:root:disk:0660
+sd*:root:disk:0660
+
+# floppy devices
+fd[01]*:root:floppy:0660
+
+# fb devices
+fb*:root:video:0660
+
+# tape devices
+ftape*:root:disk:0660
+ht*:root:disk:0660
+nftape*:root:disk:0660
+nht*:root:disk:0660
+nst*:root:disk:0660
+pt[0-9]*:root:disk:0660
+st[0-9]*:root:disk:0660
+
+# raw devices
+ram:root:disk:0640
+ram[0-9]*:root:disk:0660
+raw/*:root:disk:0660
+
+# video devices
+em8300*:root:video:0660
+
+# v4l devices
+radio*:root:video:0660
+vbi*:root:video:0660
+video*:root:video:0660
+vtx*:root:video:0660
+
+# dri devices
+dri/*:root:video:0660
+nvidia*:root:root:0666
+
+# serial+dialup devices
+ircomm*:root:ttyS:0664
+isdninfo:root:root:0444
+ppp*:root:root:0644
+ttyS*:root:ttyS:0664
+
+# memory devices
+full:root:root:0644
+kmem:root:kmem:0640
+mem:root:kmem:0640
+null:root:root:0666
+port:root:kmem:0640
+random:root:root:0644
+urandom:root:root:0644
+zero:root:root:0666
+
+# input devices
+input/event*:root:root:0660
+input/js*:root:sys:0660
+input/mice:root:root:0660
+input/mouse*:root:root:0660
+
+# joystick devices
+js*:root:sys:0660
+
+# lp devices
+irlpt*:root:ttyS:0644
+lp*:root:lp:0660
+par[0-9]*:root:lp:0660
+parport*:root:lp:0660
+usb/lp*:root:lp:0660
+
+# pty devices
+ptmx:root:tty:0666
+pty[p-za-e][0-9a-f]*:root:tty:0666
+tty[p-za-e][0-9a-f]*:root:tty:0666
+
+# usb devices
+usb/rio500:root:ttyS:0644
+usb/tty*:root:ttyS:0664
+
+# scanner devices
+usb/scanner*:root:root:0660
+
+# vc devices
+vcs*:root:tty:0620
+
+# misc devices
+rtc:root:root:0664
